pkg/kn/flags/podspec_helper.go

```diff
@@ -53,15 +52,45 @@ func containerOfPodSpec(spec *corev1.PodSpec) *corev1.Container {
 
 // UpdateEnvVars gives the configuration all the env var values listed in the given map of
 // vars.  Does not touch any environment variables not mentioned, but it can add
-// new env vars and change the values of existing ones, then sort by env key name.
-func UpdateEnvVars(spec *corev1.PodSpec, toUpdate map[string]string, toRemove []string) error {
+// new env vars and change the values of existing ones.
+func UpdateEnvVars(spec *corev1.PodSpec,
+	allArgs []string, envToUpdate *util.OrderedMap, envToRemove []string, envValueFromToUpdate *util.OrderedMap, envValueFromToRemove []string) error {
 	container := containerOfPodSpec(spec)
-	updated := updateEnvVarsFromMap(container.Env, toUpdate)
-	updated = removeEnvVars(updated, toRemove)
-	// Sort by env key name
-	sort.SliceStable(updated, func(i, j int) bool {
-		return updated[i].Name < updated[j].Name
-	})
+
+	allEnvsToUpdate := util.NewOrderedMap()
+
+	envIterator := envToUpdate.Iterator()
+	envValueFromIterator := envValueFromToUpdate.Iterator()
+
+	envKey, envValue, envExists := envIterator.NextString()
+	envValueFromKey, envValueFromValue, envValueFromExists := envValueFromIterator.NextString()
+	for _, arg := range allArgs {
+		// envs are stored as NAME=value
+		if envExists &&
+			(strings.HasPrefix(arg, envKey+"="+envValue) || strings.HasPrefix(arg, "-e="+envKey+"="+envValue) || strings.HasPrefix(arg, "--env="+envKey+"="+envValue)) {
```

Comment:
Maybe I'd extract this condition to standalone helper function. That'd allow easier unit testing eventually. Wdyt?

Reply:
Helper fn: https://github.com/knative/client/pull/1328/commits/97466727ac5bd37ce9a08e18602827fd4d4b6858#diff-5d48e95bf218a0a7a822b3f8dc2db615d9ab023ad71a01658071fca6c7a6966dR97-R108

Tests: https://github.com/knative/client/pull/1328/commits/97466727ac5bd37ce9a08e18602827fd4d4b6858#diff-88a7bb094597d334c604ecc29d3780cc31646e892d4517b3bc8c833eb833bac8R675-R765